### PR TITLE
Persistent storage type for IBM Spectrum Scale

### PR DESCRIPTION
**Pull request by alexarvinraj**
ENHANCEMENTS

- Support for persistent storage type for IBM Spectrum Scale deployment on bare metal servers.
- Spectrum Symphony version has been updated from 7.3.1 to 7.3.2.
- Spectrum Scale version has been updated from 5.1.5.1 to 5.1.7.0.
- Support public gateway creation based on existing vpc functionality.
- Support for DNS functionality.
- Updated Symphony and Scale custom images to use RHEL 8.6 instead of RHEL 8.4.
- Support custom image creation.

### CODE REVIEW

File: custom_image/storage/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+4. Run `packer build .` to create the custom image.
+
+   **Note**: The `mmbuildgpl` command's output in the Packer console (mmbuildgpl: Building GPL module completed successfully) indicates if the new custom image creation is successful. Also, the newly created image should be listed under the custom [Images of VPC](https://cloud.ibm.com/vpc-ext/compute/images) with the name that you provided for the `image_name` parameter with Active status.
+
+
+## Limitations:
+
+- The custom image creation scripts are only compatible with Red Hat-based Linux distributions operating systems.
+- The custom image creation scripts support only VSI based deployment models for both Symphony (where worker_type is specified as VSI) and Scale (where storage_type is specified as scratch).
```
Comment (AugieMena3) [inline custom_image/storage/README.md:87]: Do we have a Github issue open (and in backlog) to add custom image creation script support for deployment models that make use of bare metals (for worker nodes, and for Scale persistent storage nodes)?

File: custom_image/worker/README.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+4. Run `packer build .` to create the custom image.
+
+   **Note**: After a successful Packer build, you should be able to see a newly created custom image at [Images of VPC](https://cloud.ibm.com/vpc-ext/compute/images) with the name you provided for `image_name` parameter with Active status.
+
+
+## Limitations:
+
+- The custom image creation scripts are only compatible with Red Hat-based Linux distributions operating systems.
+- The custom image creation scripts support only VSI based deployment models for both Symphony (where worker_type is specified as VSI) and Scale (where storage_type is specified as scratch).
```
Comment (AugieMena3) [inline custom_image/worker/README.md:119]: Do we have a Github issue open (and in backlog) to add custom image creation script support for deployment models that make use of bare metals (for worker nodes, and for Scale persistent storage nodes)?

File: custom_image/worker/variables.pkr.hcl [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+variable "image_name" {
+  type        = string
+  default     = ""
+  description = "The name of the resulting custom image. To make this unique, timestamp will be appended."
+}
+
+variable "vsi_profile" {
+  type        = string
+  default     = "bx2d-2x8"
```
Comment (AugieMena3) [inline custom_image/worker/variables.pkr.hcl:33]: Does instance storage get explicitly used here?

**Review by AugieMena3 — APPROVED** [review]
Just a few questions, but looks good.

File: custom_image/storage/README.md (hunk 1 above)
Comment (alexarvinraj) [reply custom_image/storage/README.md:87]: Yes Augie, We do have.

**Pull request merged**